RxSwift/Units/Single+Maybe+Completeable.swift

```diff
@@ -0,0 +1,455 @@
+    case success(Element)
+
+    /// Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`)
+    case error(Swift.Error)
+}
+
+extension PrimitiveSequenceType where ProofType == SingleProof {
+    public typealias SingleObserver = (SingleEvent<ElementType>) -> ()
+
+    public static func create(subscribe: @escaping (@escaping SingleObserver) -> Disposable) -> PrimitiveSequence<ProofType, ElementType> {
+        let source = Observable<ElementType>.create { observer in
+            return subscribe { event in
+                switch event {
+                case .success(let element):
+                    observer.on(.next(element))
+                    observer.on(.completed)
+                case .error(let error):
+                    observer.on(.error(error))
+                }
+            }
+        }
+
+        return PrimitiveSequence(raw: source)
+    }
+
+    public func subscribe(_ observer: @escaping (SingleEvent<ElementType>) -> ()) -> Disposable {
+        var stopped = false
+        return self.primitiveSequence.asObservable().subscribe { event in
+            guard !stopped else { return }
+            stopped = true
+
+            switch event {
+            case .next(let element):
+                observer(.success(element))
+            case .error(let error):
+                observer(.error(error))
+            case .completed:
+                rxFatalError("SingleProtocol")
+            }
+        }
+    }
+}
+
+// </Single>
+
+// <Maybe>
+
+public enum MaybeEvent<Element> {
+    /// One and only sequence element is produced. (underlying observable sequence emits: `.next(Element)`, `.completed`)
+    case success(Element)
+
+    /// Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`)
+    case error(Swift.Error)
+
+    /// Sequence completed successfully.
+    case completed
+}
+
+public extension PrimitiveSequenceType where ProofType == MaybeProof {
+    public typealias MaybeObserver = (MaybeEvent<ElementType>) -> ()
+
+    public static func create(subscribe: @escaping (@escaping MaybeObserver) -> Disposable) -> PrimitiveSequence<ProofType, ElementType> {
+        let source = Observable<ElementType>.create { observer in
+            return subscribe { event in
+                switch event {
+                case .success(let element):
+                    observer.on(.next(element))
+                    observer.on(.completed)
+                case .error(let error):
+                    observer.on(.error(error))
+                case .completed:
+                    observer.on(.completed)
+                }
+            }
+        }
+
+        return PrimitiveSequence(raw: source)
+    }
+
+    public func subscribe(_ observer: @escaping (MaybeEvent<ElementType>) -> ()) -> Disposable {
+        var stopped = false
+        return self.primitiveSequence.asObservable().subscribe { event in
+            guard !stopped else { return }
```

Comment:
To be honest I'm not a big fun of 'not' statement in guard, it's really hard for me to read =) not sure about anybody else 
IMO `if stoped { return }` is much more readable